src/Symfony/Component/HttpKernel/DataCollector/DataCollectorInterface.php

```diff
@@ -24,8 +24,10 @@ interface DataCollectorInterface extends ResetInterface
 {
     /**
      * Collects data for the given Request and Response.
+     *
+     * @param \Throwable|null $exception in Symfony 5.0
      */
-    public function collect(Request $request, Response $response, \Exception $exception = null);
+    public function collect(Request $request, Response $response/*, \Exception $exception = null*/);
```

Comment:
This change means one cannot have a DataCollector compatible with Symfony 3.4, 4.4 and 5.0.
Give the ecosystem is likely supporting 3.4 + 4.x here, we'll ask everyone to drop support for 3.4 when adding support for 5.0.
One way to smoothen the issue would be to apply this very change on 3.4. WDYT?

Reply:
Opened https://github.com/symfony/symfony/pull/34230 if we move this way.